pilot/test/mock/config.go

```diff
@@ -516,12 +519,22 @@ func CheckCacheEvents(store model.ConfigStore, cache model.ConfigStoreCache, nam
 	CheckMapInvariant(store, t, namespace, n)
 
 	log.Infof("Waiting till all events are received")
-	util.Eventually(func() bool {
-		lock.Lock()
-		defer lock.Unlock()
-		return added == n && deleted == n
-
-	}, t)
+	timeout := time.After(60 * time.Second)
+	for {
+		select {
+		case <-timeout:
+			t.Fatalf("timeout waiting to receive expected events. actual added %d deleted %d. expected %d",
+				added, deleted, n)
+		default:
+			lock.Lock()
```

Comment:
@ZackButcher would you mind taking a peek at how the locks are used here ... specifically for Go-ishness?